### PR TITLE
matrix-synapse-rest-password-provider: init at unstable-2020-08-02, matrix-synapse-ldap3, matrix-synapse-pam: Add metadata

### PR DESCRIPTION
**Pull request by D4ndellion**
###### Motivation for this change
Useful coupled with matrix-corporal

Also adds metadata to the other synapse plugins

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/matrix-synapse/plugins/rest-password-provider.nix [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+buildPythonPackage rec {
+  pname = "matrix-synapse-rest-password-provider";
+  version = "unstable-2020-08-02";
+
+  src = fetchFromGitHub {
+    owner = "ma1uta";
+    repo = pname;
+    rev = "c782c84aeab1872e73b6c29aadb99d3852e26bbd";
+    sha256 = "0v9900d6919yq05s08ihrc8jclpaixfm8mwi60vhxalmddgwlhjy";
+  };
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-synapse/plugins/rest-password-provider.nix:12]: Missing meta

Comment (dali99) [reply pkgs/servers/matrix-synapse/plugins/rest-password-provider.nix:12]: no.

Comment (dali99) [reply pkgs/servers/matrix-synapse/plugins/rest-password-provider.nix:12]: sorry. Maybe? The other plugins don't have meta information either. So I assumed it's not supposed to be there

Comment (SuperSandro2000) [reply pkgs/servers/matrix-synapse/plugins/rest-password-provider.nix:12]: As far as I know this is a package so it requires a meta section.

Comment (dali99) [reply pkgs/servers/matrix-synapse/plugins/rest-password-provider.nix:12]: @abbradar was there any particular reason we didn't add meta information to the synapse plugins?

**Comment by SuperSandro2000** [comment]
ldap3 is missing a lib in inputs. Please check the others, too if they eval at all.

**Comment by dali99** [comment]
I literally only push here because I want a cloud to sync to, none of these are tested yet at _all_.

This is marked as a draft for a reason.

**Comment by dali99** [comment]
I never ended up using the rest password provider, and it's largely no longer necessary with matrix-corporal

**Pull request closed without merging**